src/gui/entry/EntryModel.cpp

```diff
@@ -318,6 +360,27 @@ QVariant EntryModel::data(const QModelIndex& index, int role) const
         if (index.column() == Paperclip) {
             return Qt::AlignCenter;
         }
+    } else if (role == Qt::ToolTipRole) {
+        if (index.column() == PasswordStrength) {
+            PasswordHealth health(entry->password());
```

Comment:
This gets recalculated on every paint event. We really need to cache this computation in `Entry` and invalidate the cache when a change is made to the entry.

Reply:
The health is now cached alongside any changes made to the password